packages/checkout/widgets-lib/src/widgets/sale/functions/sortAndDeduplicateCurrencies.test.ts

```diff
@@ -0,0 +1,197 @@
+import { SaleWidgetCurrency, SaleWidgetCurrencyType } from '../types';
+import { sortAndDeduplicateCurrencies } from './sortAndDeduplicateCurrencies';
+
+describe('sortAndDeduplicateCurrencies', () => {
+  it('sortAndDeduplicateCurrencies', () => {
+    const allCurrencies: SaleWidgetCurrency[] = [
+      {
+        base: true,
+        decimals: 6,
+        name: 'USDC',
+        address: '0x3b2d8a1931736fc321c24864bceee981b11c3c57',
+        exchangeId: 'usd-coin',
+        currencyType: SaleWidgetCurrencyType.SETTLEMENT,
+      },
+      {
+        base: false,
```

Comment:
@mimi-imtbl worth making sure that if 2 bases are returned, it will only work with 1, first one found mostlikely

Reply:
yep, fixed this edge case

Reply:
Based on the current implementation, it will pick the later `base: true`. This shouldn't be an issue, I'd assume. I'd prefer to not make this logic more complicated, will leave this for now